pkg/cmd/controller/csr.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/operator/csr"
+	"github.com/openshift/library-go/pkg/operator/events"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+const (
+	controllerName = "csr-approver-controller"
+	// Hardcoded for the Monitoring Operator:
+	// https://github.com/openshift/cluster-monitoring-operator/blob/e7a2b2e20d19ba7a9271df59e06b1fd32b8bdfe4/manifests/0000_50_cluster-monitoring-operator_03-role-binding.yaml#L25-L27
+	monitoringServiceAccountNamespace = "openshift-monitoring"
+	monitoringServiceAccountName      = "cluster-monitoring-operator"
+	monitoringCertificateSubject      = "/CN=system:serviceaccount:openshift-monitoring:prometheus-k8s"
+	monitoringLabelKey                = "metrics.openshift.io/csr.subject"
+	monitoringLabelValue              = "prometheus"
+)
+
+func RunCSRApproverController(ctx *ControllerContext) (bool, error) {
+	kubeClient, err := ctx.ClientBuilder.Client(infraClusterCSRApproverControllerServiceAccountName)
+	if err != nil {
+		return true, err
+	}
+
+	eventRecorder := events.NewRecorder(kubeClient.CoreV1().Events("test"), infraClusterCSRApproverControllerServiceAccountName, &v1.ObjectReference{})
+
+	selector := labels.NewSelector()
+	labelsRequirement, err := labels.NewRequirement(monitoringLabelKey, selection.In, []string{monitoringLabelValue})
+	if err != nil {
+		return true, err
+	}
+	selector.Add(*labelsRequirement)
```

Comment:
I believe this should be `selector = selector.Add(*labelsRequirement)`

https://github.com/enj/cluster-authentication-operator/blob/d303865704cb8d9e0e807e63d2f578ecebb42aa2/vendor/k8s.io/apimachinery/pkg/labels/selector.go#L365-L376

Reply:
:man_facepalming: Of course! You're right!

Reply:
Done